pkg/trace/obfuscate/sql.go

```diff
@@ -195,20 +200,41 @@ type tableFinderFilter struct {
 // Filter implements tokenFilter.
 func (f *tableFinderFilter) Filter(token, lastToken TokenKind, buffer []byte) (TokenKind, []byte, error) {
 	switch lastToken {
-	case From:
+	case From, Join:
 		// SELECT ... FROM [tableName]
 		// DELETE FROM [tableName]
+		// ... JOIN [tableName]
 		if r, _ := utf8.DecodeRune(buffer); !unicode.IsLetter(r) {
 			// first character in buffer is not a letter; we might have a nested
 			// query like SELECT * FROM (SELECT ...)
 			break
 		}
 		fallthrough
-	case Update, Into, Join:
+	case Update, Into:
 		// UPDATE [tableName]
 		// INSERT INTO [tableName]
-		// ... JOIN [tableName]
-		f.storeName(string(buffer))
+		if f.storeNames {
+			f.storeName(string(buffer))
+		}
+
+		if f.normalizeTables && token == ID {
```

Comment:
This doesn't seem like it belongs in the `tableFinderFilter`, but more like the `replaceFilter`.

Reply:
It requires classifying identifiers as tables and that's something we should only do once, no? I can rearrange so tablefinder comes first and returns a new `Table` token, which replacefilter acts on?

Reply:
Yeah I think this is better, I'll do that.